granary/microformats2.py

```diff
@@ -268,12 +242,10 @@ def json_to_object(mf2):
     'article': ('article', None),
   }
 
-  for h_class, mf2_type in h_class_overrides:
-    if h_class in mf2.get('type', []):
-      break  # found
-  else:
-    mf2_type = mf2util.post_type_discovery(mf2)
-
+  mf2_types = mf2.get('type') or []
+  mf2_type = ('location' if 'h-geo' in mf2_types
+              else 'person' if 'h-card' in mf2_types
+              else mf2util.post_type_discovery(mf2))
```

Comment:
`mf2util.post_type_discovery` should return 'person' for h-cards (or 'org' if the h-card has an org property that is the same as its name). I'd be more than happy to add 'h-geo' => location too.